### PR TITLE
Sockets - Jansen

### PR DESCRIPTION
**Pull request by JansenMartin**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | The model does two things: 1) It handles interactions with the database, and 2) it holds the "business logic" of the application.
Describe in your own words what the Controller is doing in Rails | The Controller does things like decoding the request payload, deciding what data to load, and figuring out which HTTP status code to send back.  (In other words, it handles any parts of the cycle related to the web server.) It also coordinates with other components inside the server.
Describe in your own words what the View is doing in Rails | The View is in charge of code that renders to the screen.
Describe an edge-case controller test you wrote | For the "destroy" and "toggle_complete" actions, I wrote tests to make sure a 404 was returned in cases where the task didn't exist.
What is the purpose of using strong params? (i.e. the params method in the controller) | It's a security feature that prevents users from accidentally updating sensitive model attributes.  Basically, we give explicit access to certain attributes that come through params, which can then be passed onto model methods.
How are Rails migrations related to Rails models? | Migrations tell the database HOW to store the data described inside the model.
Describe one area of Rails that are still unclear on | I still don't fully understand how code inside the "update" action assists with pre-populating form fields that appear via the "edit" view.

### CODE REVIEW

File: test/controllers/tasks_controller_test.rb [hunk 1]
```diff
@@ -79,50 +74,133 @@
...
+      #Arrange
+      task = Task.create!(name: "test task!", completed: true)
+
+      #Assumption
+      expect(task.completed).must_equal true
+
+      #Act
+      patch toggle_complete_task_path(task.id)
+
+      updated_task = Task.find_by(id: task.id)
```
Comment (droberts-sea) [inline test/controllers/tasks_controller_test.rb:188]: Does it go both ways? Mark complete vs incomplete?

**Comment by droberts-sea** [comment]
# Task List
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Appropriate Git Usage with no extraneous files checked in | yes
Answered comprehension questions | yes
Successfully handles: Index, Show | yes (no link to `show`, but typing in the URL works)
Index & Show tests pass | yes
Successfully handles: New, Create | yes
New & Create tests pass | yes
Successfully handles: Edit, Update | yes
Tests for Edit & Update test valid & invalid task ids | yes
Successfully handles: Destroy, Task Complete | some - it looks like you implemented `destroy`, but did not include a link so I can't try it out!
Tests for Destroy & Task Complete include tests for valid and invalid task ids | yes
Routes follow RESTful conventions | yes
Uses named routes (like `_path`) | yes
**Overall** | You're doing good work with controllers and models so far, but your views could use a little work. Your site was a little shy on styling, but more importantly you were missing some of the links to use the functionality you implemented. Other than that I like what I see - keep up the hard work!
